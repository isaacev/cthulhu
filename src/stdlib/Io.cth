intrinsic {
  fn php_print(Str) -> ();
  fn read_argv() -> [Str];
}

#[inline]
pub fn print(str: Str) -> () {
  php_print(str)
}

#[inline]
pub fn println(str: Str) -> () {
  php_print(str ++ "\n")
}

pub fn argv() -> [Str] {
  read_argv()
}
