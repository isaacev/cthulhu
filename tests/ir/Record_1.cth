use ::Io;

fn get_name(thing: { name: Str }) -> Str {
  "name: " ++ thing.name
}

#[entry]
fn main() -> () {
  let x = { names: "foo" };
  Io::println(get_name(x));
}
