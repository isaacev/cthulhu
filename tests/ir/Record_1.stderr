
  INCORRECT ARGUMENT TYPE  ---------------------------------------------------

  The function expected an argument of the type `{ name: Str }`. But the
  function call provided an argument of the type `{ names: Str }`:

     : TEST_DIR/ir/Record_1.cth
  10 | Io::println(get_name(x));
     |                      ^

